Automattic/WP-Job-Manager
# Use theme colors for job overlay, improve dashboard on small screens

### Changes Proposed in this Pull Request

* Expand the theme color detection script. Detect and calculate background and contrast colors based on the the theme.
  * Fixes color issues on dark themes, and daily chart colors when the link color is black.
* Improve job dashboard table, add column labels on mobile, improve breaking points, tweak some styles
* Fix outputting modal code multiple times

### Testing Instructions

* Enable stats
* Install a few popular themes and check that the job dashboard and the stats overlay look reasonable out of the box
* Twenty Twenty-Three, Twenty Twenty-Four has a bunch of color styles that are easy to switch in the site editor, including dark colors.
* Check that the overlay also works in the admin

### New or Updated Hooks and Templates


* `job_manager_ui_theme_support_script` - disable theme color support script

### Screenshot / Video

<img width="1284" alt="image" src="https://github.com/Automattic/WP-Job-Manager/assets/176949/a73a730a-4563-45f6-8370-32762fe4a7b8">
<img width="1286" alt="image" src="https://github.com/Automattic/WP-Job-Manager/assets/176949/c155f1ee-c60e-45f7-9768-995ba5606407">






----

| Plugin build for df2ed4356240699fa85e622a7ebadc2c53d45716 <a href="#"><img width=600></a> |
| ------------------------------------------------------------ |
| 📦 [Download plugin zip](https://wpjobmanager.com/wp-content/uploads/2024/04/wp-job-manager-zip-2810-df2ed435.zip)                       |
| ▶️ [Open in playground](https://wpjobmanager.com/playground/?core=2024/04/2810-df2ed435)             |
